lib/wab/impl/shell.rb

```diff
@@ -21,26 +19,25 @@ class Shell
 
       # Sets up the shell with the supplied configuration data.
       #
-      # config:: configuration Hash
+      # config:: Configuration object
       def initialize(config)
-        pre_path  = config['handler']['path']
-        @path_pos = pre_path.split('/').length - 1
-
-        @model    = Model.new(File.join(config['base'], config['data_dir']))
-        @type_key = config['type_key']
-
-        @verbose  = if config.has_key?('verbose')
-                      verbosity = config['verbose'].to_s.downcase
-                      if verbosity == 'true'
-                        true
-                      elsif verbosity == 'false'
-                        false
-                      end
-                    end
-
-        @http_dir    = File.expand_path(config['http']['dir'])
-        @http_port   = config['http']['port'].to_i
-        @controllers = {}
+        pre_path      = config[:path_prefix] || '/v1'
```

Comment:
might as well convert `pre_path` to an instance_var and pass it to `#start` method..

Reply:
pre_path is no longer needed after the initial creation. It makes more sense to the user to provide a prefix instead of counting though. The prefix is also used for webrick.